pybotters/models/gmocoin.py

```diff
@@ -0,0 +1,616 @@
+
+    def _onmessage(self, mes: OrderBook) -> None:
+        data = mes["asks"] + mes["bids"]
+        result = self.find({"symbol": mes["symbol"]})
+        self._delete(result)
+        self._insert(cast(List[Item], data))
+
+
+class OrderStore(DataStore):
+    syncroot = asyncio.Lock()
+    _KEYS = ["order_id"]
+
+    def _onresponse(self, data: List[Order]) -> None:
+        self._insert(cast(List[Item], data))
+
+    async def _onmessage(self, mes: Order) -> None:
+        async with self.syncroot:
+            if mes["order_status"] in (OrderStatus.WAITING, OrderStatus.ORDERED):
+                self._update([cast(Item, mes)])
+            else:
+                self._delete([cast(Item, mes)])
+
+    async def _onexecution(self, mes: Execution) -> None:
+        async with self.syncroot:
+            current = cast(Order, self.get({"order_id": mes["order_id"]}))
+            if (
+                mes["order_executed_size"]
+                and current
+                and current["executed_size"] < mes["order_executed_size"]
+            ):
+                current["executed_size"] = mes["order_executed_size"]
+                remain = current["size"] - current["executed_size"]
+                if remain == 0:
+                    self._delete([cast(Item, current)])
+                else:
+                    self._update([cast(Item, current)])
+
+
+class ExecutionStore(DataStore):
+    _KEYS = ["execution_id"]
+
+    def sorted(self, query: Item = {}) -> List[Execution]:
+        result = []
+        for item in self:
+            if all(k in item and query[k] == item[k] for k in query):
+                result.append(item)
+        result.sort(key=lambda x: x["execution_id"], reverse=True)
+        return result
+
+    def _onresponse(self, data: List[Execution]) -> None:
+        self._insert(cast(List[Item], data))
+
+    async def _onmessage(self, mes: Execution) -> None:
+        self._insert([cast(Item, mes)])
+
+
+class PositionStore(DataStore):
+    _KEYS = ["position_id"]
+
+    def _onresponse(self, data: List[Position]) -> None:
+        self._update(cast(List[Item], data))
+
+    def _onmessage(self, mes: Position, type: MessageType) -> None:
+        if type == MessageType.OPR:
+            self._insert([cast(Item, mes)])
+        elif type == MessageType.CPR:
+            self._delete([cast(Item, mes)])
+        else:
+            self._update([cast(Item, mes)])
+
+
+class PositionSummaryStore(DataStore):
+    _KEYS = ["symbol", "side"]
+
+    def _onresponse(self, data: List[PositionSummary]) -> None:
+        self._update(cast(List[Item], data))
+
+    def _onmessage(self, mes: PositionSummary) -> None:
+        self._update([cast(Item, mes)])
+
+
+class MessageHelper:
+    @staticmethod
+    def to_tickers(data: List[Item]) -> List["Ticker"]:
+        return [MessageHelper.to_ticker(x) for x in data]
+
+    @staticmethod
+    def to_ticker(data: Item) -> "Ticker":
+        return Ticker(
+            ask=int(data["ask"]),
```

Comment:
様々な箇所で価格(price)などのフィールドを `int` で定義されていますがBTC以外の銘柄(XRPなど)は浮動小数の文字列データとなっているので変換時 `ValueError` になります。 `int` ではなく `Decimal` でいいかと思います。
```suggestion
            ask=Decimal(data["ask"]),
```

Reply:
なるほど、 `BTC_JPY` しか触っておりませんでした。Decimal に変更で良いと思います。